### PR TITLE
Include list of dependencies in documentation

### PR DESCRIPTION
**Pull request by ludovicsteinbach**
To address https://github.com/ansys/grantami-recordlists/issues/283, include a list of all dependencies for all versions of the metapackage.

Heavily based on https://github.com/ansys/pyansys/pull/693 and https://github.com/ansys/pyansys/pull/702. Simplified in places with assumptions we can make for PyGranta packages.

Packages like RecordLists will then be able to link to this page, providing a mapping between Ansys versions and pygranta packages.

Screenshot of the output:
image

One weird outcome of this is that the auto-generated libraries include `https://grantami.docs.pyansys.com/` as their documentation.

### CODE REVIEW

File: doc/_package_list.py [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+                "version": library_version,
+                "docs": doc_link,
+                "pypi": pypi_link,
+            }
+        )
+    return pyansys_libraries
+
+
+def get_release_branches_in_metapackage():
+    """Retrieve the release branches in the PyAnsys metapackage."""
```
Comment (Andy-Grigg) [inline doc/_package_list.py:50]: Should this be PyGranta, and the comment below?

File: doc/source/conf.py [hunk 2]
```diff
@@ -97,3 +91,23 @@
 
 # Ignore ansys links for linkcheck
 linkcheck_ignore = ["https://www.ansys.com/"]
+
+# sphinx-jinja configuration
+jinja_contexts = {"package_versions_ctx": {}}
+
+########
+# Fetch all versions of the metapackage and dependencies
```
Comment (Andy-Grigg) [inline doc/source/conf.py:99]:
```suggestion
# Fetch all versions of the metapackage and Ansys dependencies
```

I think?

**Review by Andy-Grigg — APPROVED** [review]
The code looks fine, just a couple of possibly-incorrect comments. The output also looks good.

As far as the openapi links go, we could leave the docs as they are, or alternatively:

* Change the documentation link in PyPI, although I'm not sure what else we want to link to
* Ignore packages with openapi in the name
* Remove the docs for pacakges with openapi in the name
*

**Comment by ludovicsteinbach** [comment]
Re-requested a review as I made some logic changes.

It now filters out auto-generated openapi packages, and the generate page does not include a link for them at all:
image

**Review by Andy-Grigg — APPROVED** [review]

**Pull request merged**